lib/pure/times.nim

```diff
@@ -1319,83 +1345,128 @@ when not defined(JS):
-    result.second = t.getSeconds()
-    result.minute = t.getMinutes()
-    result.hour = t.getHours()
-    result.monthday = t.getDate()
-    result.month = Month(t.getMonth())
-    result.year = t.getFullYear()
-    result.weekday = weekDays[t.getDay()]
-    result.timezone = getTimezone()
-
-    result.yearday = result.monthday - 1
-    for month in mJan..<result.month:
-      result.yearday += getDaysInMonth(month, result.year)
-
-  proc getGMTime(t: Time): TimeInfo =
-    result.second = t.getUTCSeconds()
-    result.minute = t.getUTCMinutes()
-    result.hour = t.getUTCHours()
-    result.monthday = t.getUTCDate()
-    result.month = Month(t.getUTCMonth())
-    result.year = t.getUTCFullYear()
-    result.weekday = weekDays[t.getUTCDay()]
-
-    result.yearday = result.monthday - 1
-    for month in mJan..<result.month:
-      result.yearday += getDaysInMonth(month, result.year)
-
-  proc timeInfoToTime(timeInfo: TimeInfo): Time = toTime(timeInfo)
-
-  proc toTime*(timeInfo: TimeInfo): Time = newDate($timeInfo)
-
-  proc `-` (a, b: Time): int64 =
-    return a.getTime() - b.getTime()
+proc fromSeconds*(since1970: int64): Time {.tags: [], raises: [], benign, deprecated.} =
+  ## Takes an int which contains the number of seconds since the unix epoch and
+  ## returns a time object.
+  Time(since1970)
 
-  var
-    startMilsecs = getTime()
+proc toSeconds*(time: Time): float {.tags: [], raises: [], benign, deprecated.} =
+  ## Returns the time in seconds since the unix epoch.
+  float(time)
 
-  proc getStartMilsecs(): int =
-    ## get the milliseconds from the start of the program
-    return int(getTime() - startMilsecs)
+proc getLocalTime*(time: Time): DateTime {.tags: [], raises: [], benign, deprecated.} =
+  ## Converts the calendar time `time` to broken-time representation,
+  ## expressed relative to the user's specified time zone.
+  time.local
 
-  proc fromSeconds(since1970: float): Time = result = newDate(since1970 * 1000)
+proc getGMTime*(time: Time): DateTime {.tags: [], raises: [], benign, deprecated.} =
+  ## Converts the calendar time `time` to broken-down time representation,
+  ## expressed in Coordinated Universal Time (UTC).
+  time.utc
 
-  proc toSeconds(time: Time): float = result = time.getTime() / 1000
+proc getTimezone*(): int {.tags: [TimeEffect], raises: [], benign, deprecated.} =
+  ## returns the offset of the local (non-DST) timezone in seconds west of UTC.
+  when defined(JS):
+    return newDate().getTimezoneOffset() * 60
+  elif defined(freebsd) or defined(netbsd) or defined(openbsd):
+    var a = timec(nil)
+    let lt = localtime(addr(a))
+    # BSD stores in `gmtoff` offset east of UTC in seconds,
+    # but posix systems using west of UTC in seconds
+    return -(lt.gmtoff)
+  else:
+    return timezone
+
+proc timeInfoToTime*(dt: DateTime): Time {.tags: [], benign, deprecated.} =
+  ## Converts a broken-down time structure to
+  ## calendar time representation. The function ignores the specified
```

Comment:
neither weekdays nor yeardays are computed here, so the comment about ignoring and recomputing them seems to be irrelevant.